### DISCUSSION TITLE
iamge magick 7 ImageMagick-7.1.0-Q16-HDRI crop xy not working

### DISCUSSION THREAD

**Discussion by marchenault** [ImageMagick/ImageMagick#4218] [Help]
I'm trying to crop out an image that is 180x160 at a specific x&y after rotation and resizing with the following magick command

input.jpg -resize 135x137 -background white -rotate 18 -crop 160x180-37-53 -sharpen 0x.2 +repage -quality 92  output.jpg

In the attached
![interface](https://user-images.githubusercontent.com/73128308/133159466-8c9f8144-5809-49ca-bf93-4bd7db18e64f.png)

The blue frame on the left represents what the image should look like when converted (on a white background). Instead I get the image on the right but even then it is not 160x180 and the xy coordinates of the crop are not applied. 

the input:
![input](https://user-images.githubusercontent.com/73128308/133159568-b99ae3ca-a7bd-4db3-a012-6b88c110766d.jpg)

The output:
![output](https://user-images.githubusercontent.com/73128308/133159539-9d74f6b1-0de4-4f08-8084-768499376e3e.jpg)

the strange part is that if the image is scaled up it seems to work file.

**Comment by snibgo** [comment]
I get the result you want if I insert "+repage" after the rotate.

**Comment by marchenault** [comment]
I don't 

input.jpg -resize 135x137 -background white -rotate 24 +repage -crop 160x180-32-37 -sharpen 0x.2 -quality 92 ouput.jpg

gives me 

![ouput2](https://user-images.githubusercontent.com/73128308/133163275-1b04acc9-d96e-447c-9559-9fd2d86cca25.jpg)

the image should be 160x180 and there should be a white border top and left.

**Comment by snibgo** [comment]
Ah, okay. But after the rotate, there is no white border at the top and left of the image, so where do think those borders should come from? They won't come from the "-crop". See https://imagemagick.org/script/command-line-options.php#crop and https://imagemagick.org/script/command-line-processing.php#geometry . 

But they can come from "-extent". For clarity, I use a blue background. Windows BAT script:
```
magick ^
  133159568-b99ae3ca-a7bd-4db3-a012-6b88c110766d.jpg ^
  -resize 135x137 ^
  -background blue -rotate 24 +repage ^
  -extent "160x180-32-37^!" +repage ^
  -sharpen 0x.2 -quality 92 ^
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/133171846-c9788796-ee74-41e8-a3b6-81b5332637c8.png)

**Comment by marchenault** [comment]
I'm working with what you added above but depending on the resize of the image and the rotation etc I get very different outcomes.

Is there a way to modify an image, rotate, resize etc and then paste it onto a new 160x180 white image (160x180) at X & Y coordinates?

**Comment by fmw42** [comment]
You can use -distort SRT to rotate, scale and translate. Or just rotate, then scale as desired. Then crop the area you want. The use -geometry +X+Y -compose over -composite to put the crop over the background that you want. Or just use -extent with some color for the background and you can essentially position it where desired.

**Comment by snibgo** [comment]
> Is there a way to modify an image, rotate, resize etc and then paste it onto a new 160x180 white image (160x180) at X & Y coordinates?

Yes. The command does exactly that. More precisely, it resizes and rotates, and places the top-left corner of the resulting image at the location (32,37). To show this more clearly, we use a blue background for the rotate, and a green background for the extent:
```
magick ^
  133159568-b99ae3ca-a7bd-4db3-a012-6b88c110766d.jpg ^
  -resize 135x137 ^
+write x0.png ^
  -background blue -rotate 24 +repage ^
+write x1.png ^
  -background green -extent "160x180-32-37^!" +repage ^
+write x2.png ^
  -sharpen 0x.2 -quality 92 ^
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/133184571-3c40f8e9-273c-424e-84c2-40eb178f393c.png)

The top-left blue corner is at (32,37). I write to x0.png etc so you can see the different stages.

If that isn't what you want, then you need to be more precise about what you do want.

**Comment by marchenault** [comment]
Got it. Thank you for your help. Works great. just needed to redo my calculations for X % Y.
